### ISSUE TITLE
difftest.cpp:40: void init_difftest(rtlreg_t*): Assertion `handle' failed.

### ISSUE THREAD

**Issue by iam41xin** [OSCPU/NutShell#10]
Hi mates,

While I run the simulation by 'make emu', I met 2 issues:

make[1]: Leaving directory '/NutShell/build/emu-compile'
Using seed = 7348
The image is ready-to-run/linux.bin
[warning] sdcard img not found
emu: /NutShell/src/test/csrc/difftest.cpp:40: void init_difftest(rtlreg_t*): Assertion `handle' failed.
make: *** [Makefile:100: emu] Aborted

As above shown, there is a warning that image files were not found, this issue has been solved after I modified the sdcard.cpp in folder $NUTSHELL_HOME/src/test/csrc/.
But, the 2nd issue still couldn't be solved, in Makefile, default REF_SO has been setup correctly, why assertion 'handle' still failed?

Thanks!

**Comment by wakafa1** [comment]
Usually mismatched NEMU can cause such error. I just cloned and built this project from scratch but cannot reproduce the problem you mentioned. Are you using the ready-to-run mode? Or did you compile a copy of nemu yourself?

**Comment by iam41xin** [comment]
Hi Wakafa,
Initially, what I've done is nothing but to clone the git, download and install mill and run make && make emu in folder $NUTSHELL_HOME.
I've tried to set $(USE_READY_TO_RUN_NEMU) option from true to false and clone nemu, after setting NEMU_HOME, 'make emu' compiled the file riscv64-nemu-interpreter-so successfully in NEMU_HOME/build, but still, Assertion `handle' failed.
Also I compiled microbench-riscv64-nutshell.bin with nexus-am successfully, using this file to replace linux.bin, I got the same result, Assertion `handle' failed.
My OS is Linux CentOS 7, x86_64.
